In [49]:
import pandas as pd
import numpy as np
pd.set_option('future.no_silent_downcasting', True)

In [50]:
df_water = pd.read_excel("data/SuEatableLife_Food_Fooprint_database.xlsx", sheet_name = "SEL WF for users")
df_water = df_water.replace('-',np.nan)
df_co2 = pd.read_csv("data/Clean_CO2.csv")
df_prod = pd.read_csv("data/Production_Crops_Livestock_E_All_Data.csv")

C:\Users\User\AppData\Local\Temp\ipykernel_10140\211997447.py:4: DtypeWarning: Columns (11,14,17,20,23,26,29,32,35,38,41,44,47,50,53,56,59,62,65,68,71,74,77,80,83,86,89,92,95,98,101,104,107,110,113,116,119,122,125,128,131,134,137,140,143,146,149,152,155,158,161,164,167,170,173,176,179,182,185,188,191,194,197) have mixed types. Specify dtype option on import or set low_memory=False.
  df_prod = pd.read_csv("data/Production_Crops_Livestock_E_All_Data.csv")


In [51]:
df_water.rename(columns={ 
                       "FOOD COMMODITY GROUP" : "food_commodity_group",
                       "Food commodity ITEM" : "food_commodity_item",  
             "Food commodity TYPOLOGY" : "food_commodity_typology",
             "Food commodity sub-TYPOLOGY": "Food_commodity_sub-TYPOLOGY",
                }, inplace= True)

In [52]:
print(len(df_co2["food_commodity_item"]))
print(len(df_water["food_commodity_item"]))

324
320


In [53]:
#lower all strings
df_water = df_water.map(lambda x: x.lower() if type(x) == str else x)
df_co2 = df_co2.map(lambda x: x.lower() if type(x) == str else x)
df_prod = df_prod.map(lambda x: x.lower() if type(x) == str else x)

In [54]:
def remove_asterix(df):
    '''
    We remove * as it corresponds to typologies having only one item in it and as it is different between the 2 datasets
    '''
    df.replace(to_replace = r"\*+",value=  '',inplace = True,regex=True)
    return df
def remove_s(df):
    '''
    Removing s at the end of words to avoid bananas not matching with banana
    '''
    df.replace({r'(\w+)(s)$' : r'\1'}, inplace = True,regex=True)
    return df


In [55]:
df_co2 = remove_asterix(df_co2)
df_water = remove_asterix(df_water)
df_water['food_commodity_item'] = remove_s(df_water['food_commodity_item'])
df_co2['food_commodity_item'] = remove_s(df_co2['food_commodity_item'])
df_prod['Item'] = remove_s(df_prod['Item'])

In [56]:
#remove lines where it is imported 
def remove_parenthesis_in_item(df, content_of_parenthesis):
    '''
    At the end of certain items there are parenthesis giving information on the item e.g (i) correspond to imported.
    However the others datasets don't have the same level of information and we just keep the main item if existing.
    '''
    df_imported = df[df['food_commodity_item'].str.contains(fr"\({content_of_parenthesis}\)")]
    for idx,row in df_imported.iterrows():
        item_to_find = row["food_commodity_item"][:-4]
        if df['food_commodity_item'].str.contains(item_to_find).sum() >1:
            df = df[df["food_commodity_item"] != row["food_commodity_item"]]
        else:
            # Replace using loc instead of replace to avoid warnings
            mask = df["food_commodity_item"] == row["food_commodity_item"]
            df.loc[mask, "food_commodity_item"] = item_to_find
    return df

# We remove those detailed informations but we might create small games to learn what consumes more between greenhouse or imported for example
df_co2 = remove_parenthesis_in_item(df_co2, "g")
df_co2 = remove_parenthesis_in_item(df_co2, "i")
df_water = remove_parenthesis_in_item(df_water, "f")
df_co2 = remove_parenthesis_in_item(df_co2, "G")
df_co2 = remove_parenthesis_in_item(df_co2, "f")
df_co2 = remove_parenthesis_in_item(df_co2, "l")
df_co2

C:\Users\User\AppData\Local\Temp\ipykernel_10140\2638439649.py:10: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  if df['food_commodity_item'].str.contains(item_to_find).sum() >1:


Unnamed: 0    food_commodity_group               food_commodity_item  \
0             0  agricultural processed                       beer in can   
1             1  agricultural processed                      beer in glas   
2             2  agricultural processed                  beer modular can   
3             3  agricultural processed  chocolate or cream filled cookie   
4             4  agricultural processed                     simple cookie   
..          ...                     ...                               ...   
319         319                 fishing                            octopu   
320         320                 fishing                             squid   
321         321                 fishing                     prawns/shrimp   
322         322                 fishing                           octopus   
323         323                 fishing                    prawns/shrimps   

     CF_food_item uncertainty          Suggested CF value  \
0        0.693000           l                     ok item   
1        0.960500           l                     ok item   
2        0.530600           l                     ok item   
3        1.533000           l                     ok item   
4        1.390750           l                     ok item   
..            ...         ...                         ...   
319      5.800000           h  better typol. or subtypol.   
320      6.910000           h  better typol. or subtypol.   
321      7.040000           l                     ok item   
322      7.878611           h             better typology   
323     15.065000           l                     ok item   

    food_commodity_typology  CF_food_typology  \
0                      beer              0.69   
1                      beer              0.69   
2                      beer              0.69   
3                  biscuits              1.47   
4                  biscuits              1.47   
..                      ...               ...   
319               shellfish              5.67   
320               shellfish              5.67   
321               shellfish              5.67   
322        shellfish frozen             11.47   
323        shellfish frozen             11.47   

               Food_commodity_sub-TYPOLOGY  CF_food_sub_typology  
0                                      NaN                   NaN  
1                                      NaN                   NaN  
2                                      NaN                   NaN  
3                                      NaN                   NaN  
4                                      NaN                   NaN  
..                                     ...                   ...  
319                   cephalopod (sub-typ)                  5.80  
320                   cephalopod (sub-typ)                  5.80  
321  crustacean (sub-typ) (prawns/shrimps)                  7.04  
322                                    NaN                   NaN  
323                                    NaN                   NaN  

[289 rows x 10 columns]

In [57]:
#TODO : rendre ça moins cheum
pd.options.mode.chained_assignment = None 

# First, let's identify the exact columns in each dataframe
co2_columns = df_co2.columns.tolist()
water_columns = df_water.columns.tolist()

# 1. Create indicator column during merge
co2_and_water_full = df_co2.merge(df_water, 
                            on=["food_commodity_item", "food_commodity_group"], 
                            how='outer',
                            indicator=True)

# 2. Get rows common to both dataframes
co2_and_water = co2_and_water_full[co2_and_water_full['_merge'] == 'both'].drop(columns=['_merge'])

# 3. Get rows unique to df_co2 - using only columns that actually exist in df_co2
df_only_co2 = df_co2[~df_co2.set_index(["food_commodity_item", "food_commodity_group"])
                      .index.isin(co2_and_water.set_index(["food_commodity_item", "food_commodity_group"])
                      .index)]

# 4. Get rows unique to df_water - using only columns that actually exist in df_water
df_only_water = df_water[~df_water.set_index(["food_commodity_item", "food_commodity_group"])
                        .index.isin(co2_and_water.set_index(["food_commodity_item", "food_commodity_group"])
                        .index)] 

df_asterisk_water = df_only_water[df_only_water["Suggested WF value"]=="item matching typology"]
df_asterisk_co2 =  df_only_co2[df_only_co2["Suggested CF value"]=="Item matching typology"]
df_different_co2 = df_only_co2.loc[df_only_co2.index.difference(df_asterisk_co2.index)]
df_different_water = df_only_water.loc[df_only_water.index.difference(df_asterisk_water.index)]

#Merge typologies with unique item in water with
for index,row in df_asterisk_water.iterrows():
    if row["food_commodity_typology"] in list(df_different_co2["food_commodity_typology"]):
        element = df_only_co2[df_only_co2["food_commodity_typology"]==row["food_commodity_typology"]]
        line_to_add = element
        
        line_to_add["CF_food_item"] = list(element["CF_food_typology"])[0]
        line_to_add["food_commodity_item"] = list(element["food_commodity_typology"])[0]
        #removing the detailed elements and keeping only the typology to get the same as for water 
        df_co2 = df_co2[df_co2["food_commodity_typology"]!=row["food_commodity_typology"]]
        df_co2 = pd.concat([df_co2, line_to_add], ignore_index=True)


In [58]:
def remove_item(df, item):
    return df[df["food_commodity_item"]!=item]
df_co2 = remove_item(df_co2, "cheese")
df_water = remove_item(df_water, "cheese")
#Remove stocks as it is just animals being moved and not production
df_prod = df_prod[df_prod["Element"]!="stocks"]
df_prod

Area Code Area Code (M49)                                     Area  \
0              2            '004                              afghanistan   
1              2            '004                              afghanistan   
2              2            '004                              afghanistan   
3              2            '004                              afghanistan   
4              2            '004                              afghanistan   
...          ...             ...                                      ...   
78167       5817            '902  net food importing developing countries   
78168       5817            '902  net food importing developing countries   
78169       5817            '902  net food importing developing countries   
78170       5817            '902  net food importing developing countries   
78171       5817            '902  net food importing developing countries   

       Item Code Item Code (CPC)  \
0            221          '01371   
1            221          '01371   
2            221          '01371   
3            711          '01654   
4            711          '01654   
...          ...             ...   
78167       1729          'f1729   
78168       1729          'f1729   
78169       1735          'f1735   
78170       1735          'f1735   
78171       1735          'f1735   

                                                    Item  Element Code  \
0                                      almonds, in shell          5312   
1                                      almonds, in shell          5412   
2                                      almonds, in shell          5510   
3      anise, badian, coriander, cumin, caraway, fenn...          5312   
4      anise, badian, coriander, cumin, caraway, fenn...          5412   
...                                                  ...           ...   
78167                                    treenuts, total          5412   
78168                                    treenuts, total          5510   
78169                                 vegetables primary          5312   
78170                                 vegetables primary          5412   
78171                                 vegetables primary          5510   

              Element   Unit       Y1961  ... Y2020N       Y2021  Y2021F  \
0      area harvested     ha         0.0  ...    NaN     36862.0       a   
1               yield  kg/ha         NaN  ...    NaN      1743.2       a   
2          production      t         0.0  ...    NaN     64256.0       a   
3      area harvested     ha         NaN  ...    NaN     25357.0       e   
4               yield  kg/ha         NaN  ...    NaN       705.0       e   
...               ...    ...         ...  ...    ...         ...     ...   
78167           yield  kg/ha       614.5  ...    NaN       683.0       e   
78168      production      t    255724.0  ...    NaN   2667688.9       e   
78169  area harvested     ha   1881425.0  ...    NaN   7753692.0       e   
78170           yield  kg/ha      6617.1  ...    NaN     12136.8       e   
78171      production      t  12449526.0  ...    NaN  94105040.2       e   

      Y2021N        Y2022  Y2022F Y2022N        Y2023  Y2023F Y2023N  
0        NaN     36462.00       a    NaN     37000.00       a    NaN  
1        NaN      1742.00       a    NaN      1810.80       a    NaN  
2        NaN     63515.00       a    NaN     67000.00       a    NaN  
3        NaN     25403.00       e    NaN     25439.00       e    NaN  
4        NaN       704.00       e    NaN       704.00       e    NaN  
...      ...          ...     ...    ...          ...     ...    ...  
78167    NaN       676.60       e    NaN       682.70       e    NaN  
78168    NaN   2854981.94       e    NaN   2812938.79       e    NaN  
78169    NaN   7754677.00       e    NaN   7795058.00       e    NaN  
78170    NaN     12442.00       e    NaN     12396.10       e    NaN  
78171    NaN  96483623.71       e    NaN  96628287.75       e    N

In [59]:
old_to_new_typo = pd.read_excel("data/old_to_new_typo.xlsx")
from_old_typo_to_new_typo = dict(zip(old_to_new_typo['old_typo'], old_to_new_typo['new_typo']))
df_co2["food_commodity_typology"] = df_co2["food_commodity_typology"].replace(from_old_typo_to_new_typo)
df_water["food_commodity_typology"] = df_water["food_commodity_typology"].replace(from_old_typo_to_new_typo)
df_water.loc[df_water["food_commodity_item"]=="palm nut","food_commodity_typology"] = "nuts"
df_co2

def avg_of_typology(df, col_to_avg):
    df[col_to_avg] = df.groupby("food_commodity_typology")[col_to_avg].transform('mean')
    df[col_to_avg] = round(df[col_to_avg], 2)
    return df
df_co2 = avg_of_typology(df_co2, "CF_food_typology")
df_water = avg_of_typology(df_water, "Water Footprint cc water/g o cc of food TYPOLOGY")
df_water

food_commodity_group food_commodity_item  \
0    agricultural processed         apple juice   
1    agricultural processed        barley flake   
2    agricultural processed               beans   
3    agricultural processed                beer   
4    agricultural processed          beet sugar   
..                      ...                 ...   
315                 fishing            sturgeon   
316                 fishing             tilapia   
317                 fishing               trout   
318                 fishing              turbot   
319                 fishing       wuchang bream   

     Water Footprint liters water/kg o liter of food ITEM  \
0                                               1140.0      
1                                               1977.0      
2                                                801.0      
3                                                155.0      
4                                                865.0      
..                                                 ...      
315                                            13420.0      
316                                             2263.0      
317                                             2188.0      
318                                             1504.0      
319                                            10550.0      

    Uncertainty    low  (L)  high (H)      Suggested WF value  \
0                                   l                 ok item   
1                                   l                 ok item   
2                                   l                 ok item   
3                                   h  item matching typology   
4                                   l                 ok item   
..                                ...                     ...   
315                                 l                 ok item   
316                                 l                 ok item   
317                                 l                 ok item   
318                                 l                 ok item   
319                                 l                 ok item   

    food_commodity_typology  Water Footprint cc water/g o cc of food TYPOLOGY  \
0                     drink                                            824.86   
1         breakfast cereals                                           2196.00   
2         vegetables frozen                                            750.00   
3                     drink                                            824.86   
4                     sugar                                           1294.00   
..                      ...                                               ...   
315                    fish                                           2313.00   
316                    fish                                           2313.00   
317                    fish                                           2313.00   
318                    fish                                           2313.00   
319                    fish                                           2313.00   

    Food_commodity_sub-TYPOLOGY  \
0                           NaN   
1                           NaN   
2                           NaN   
3                           NaN   
4                           NaN   
..                          ...   
315                         NaN   
316                         NaN   
317                         NaN   
318                         NaN   
319                         NaN   

     Water Footprint cc water/g o cc of food sub-TYPOLOGY  
0                                                  NaN     
1                                                  NaN     
2                                                  NaN     
3                                                  NaN     
4                                                  NaN     
..                                                 ...     
315                                                NaN     
316   

In [60]:
# Your original inner merge
co2_and_water = df_co2.merge(df_water, on = ["food_commodity_item","food_commodity_group","food_commodity_typology"], how='inner')
co2_and_water

Unnamed: 0    food_commodity_group               food_commodity_item  \
0             3  agricultural processed  chocolate or cream filled cookie   
1             4  agricultural processed                     simple cookie   
2             6  agricultural processed                 bread multicereal   
3             7  agricultural processed                       bread plain   
4             8  agricultural processed                       bread whole   
..          ...                     ...                               ...   
132           0  agricultural processed                              beer   
133           1  agricultural processed                              beer   
134           2  agricultural processed                              beer   
135          95  agricultural processed                              wine   
136          96  agricultural processed                              wine   

     CF_food_item uncertainty Suggested CF value food_commodity_typology  \
0         1.53300           l            ok item                biscuits   
1         1.39075           l            ok item                biscuits   
2         0.70300           h    better typology                   bread   
3         0.88500           l            ok item                   bread   
4         0.77200           l            ok item                   bread   
..            ...         ...                ...                     ...   
132       0.69000           l            ok item                   drink   
133       0.69000           l            ok item                   drink   
134       0.69000           l            ok item                   drink   
135       0.66000           l            ok item                   drink   
136       0.66000           h    better typology                   drink   

     CF_food_typology Food_commodity_sub-TYPOLOGY_x  CF_food_sub_typology  \
0                1.47                           NaN                   NaN   
1                1.47                           NaN                   NaN   
2                0.98                           NaN                   NaN   
3                0.98                           NaN                   NaN   
4                0.98                           NaN                   NaN   
..                ...                           ...                   ...   
132              1.47                           NaN                   NaN   
133              1.47                           NaN                   NaN   
134              1.47                           NaN                   NaN   
135              1.47                           NaN                   NaN   
136              1.47                           NaN                   NaN   

     Water Footprint liters water/kg o liter of food ITEM  \
0                                               2902.0      
1                                               1723.0      
2                                                771.0      
3                                               1031.0      
4                                                887.0      
..                                                 ...      
132                                              155.0      
133                                              155.0      
134                                              155.0      
135                                              524.0      
136                                              524.0      

    Uncertainty    low  (L)  high (H)      Suggested WF value  \
0                                   l                 ok item   
1                                   l                 ok item   
2                                   l         better typology   
3                                   l                 ok item   
4                                   l                 ok item   
..                                ...                     ...   
132                                 h  item matching t

## Transform to hierarchical json


In [61]:

import json

def convert_df_to_hierarchical_json(df,value):
    # Create the root node with 'children' for each unique food_commodity_group
    result = {"name": "Food", "children": []}
    
    # Group by the first level (food_commodity_group)
    for group_name, group_df in df.groupby('food_commodity_group'):
        group_node = {"name": group_name, "children": []}
        
        # Group by the second level (food_commodity_typology)
        for typology_name, typology_df in group_df.groupby('food_commodity_typology'):
            typology_node = {"name": typology_name, "children": []}
            
            # Group by the third level (food_commodity_item)
            for item_name, item_df in typology_df.groupby('food_commodity_item'):
                # If there are multiple CF_food_item values for a single item, 
                # we take the average or any other aggregation you prefer
                cf_value = item_df[value].mean()
                
                item_node = {"name": item_name, "value": round(float(cf_value),2)}
                typology_node["children"].append(item_node)
            
            group_node["children"].append(typology_node)
        
        result["children"].append(group_node)
    
    return json.dumps(result, indent=2)

co2_json = convert_df_to_hierarchical_json(co2_and_water,'CF_food_item')
# Writing to sample.json
with open("data/data_carbon.json", "w") as outfile:
    outfile.write(co2_json)
water_json = convert_df_to_hierarchical_json(co2_and_water,'Water Footprint liters water/kg o liter of food ITEM')
# Writing to sample.json
with open("data/data_water.json", "w") as outfile:
    outfile.write(water_json)


There are 135 items in common.
Problem to resolve : different orders of words and subtlities between products.

## Merging with df_prod (unfinished)

In [62]:
# Your original inner merge
all_df_merged = co2_and_water.merge(df_prod, left_on = "food_commodity_item",right_on= "Item",  how='inner')
all_df_merged

Unnamed: 0    food_commodity_group food_commodity_item  CF_food_item  \
0             64  agricultural processed           olive oil         3.265   
1             64  agricultural processed           olive oil         3.265   
2             64  agricultural processed           olive oil         3.265   
3             64  agricultural processed           olive oil         3.265   
4             64  agricultural processed           olive oil         3.265   
...          ...                     ...                 ...           ...   
8437          96  agricultural processed                wine         0.660   
8438          96  agricultural processed                wine         0.660   
8439          96  agricultural processed                wine         0.660   
8440          96  agricultural processed                wine         0.660   
8441          96  agricultural processed                wine         0.660   

     uncertainty Suggested CF value food_commodity_typology  CF_food_typology  \
0              l            ok item                     oil              2.11   
1              l            ok item                     oil              2.11   
2              l            ok item                     oil              2.11   
3              l            ok item                     oil              2.11   
4              l            ok item                     oil              2.11   
...          ...                ...                     ...               ...   
8437           h    better typology                   drink              1.47   
8438           h    better typology                   drink              1.47   
8439           h    better typology                   drink              1.47   
8440           h    better typology                   drink              1.47   
8441           h    better typology                   drink              1.47   

     Food_commodity_sub-TYPOLOGY_x  CF_food_sub_typology  ...  \
0                              NaN                   NaN  ...   
1                              NaN                   NaN  ...   
2                              NaN                   NaN  ...   
3                              NaN                   NaN  ...   
4                              NaN                   NaN  ...   
...                            ...                   ...  ...   
8437                           NaN                   NaN  ...   
8438                           NaN                   NaN  ...   
8439                           NaN                   NaN  ...   
8440                           NaN                   NaN  ...   
8441                           NaN                   NaN  ...   

                 Y2020N      Y2021 Y2021F             Y2021N      Y2022  \
0                   NaN    1043.00      i                NaN    1050.00   
1     unofficial figure   16500.00      x  unofficial figure   15500.00   
2     unofficial figure   71000.00      x  unofficial figure   88200.00   
3     unofficial figure   30000.00      x  unofficial figure   33000.00   
4     unofficial figure   22600.00      x  unofficial figure   20000.00   
...                 ...        ...    ...                ...        ...   
8437                NaN    7787.29      e                NaN    7279.47   
8438                NaN  378076.85      e                NaN  378175.21   
8439                NaN   16228.00      e                NaN   14214.00   
8440                NaN   27202.37      e                NaN   26719.03   
8441                NaN  182518.33      e                NaN  174758.93   

      Y2022F             Y2022N Y2023 Y2023F  Y2023N  
0          i                NaN   NaN    NaN     NaN  
1          x  unofficial figure   NaN    NaN     NaN  
2          x  unofficial figure   NaN    NaN     NaN  
3          a                NaN   NaN    NaN     NaN  
4          x  unofficial figure   NaN    NaN     NaN  
...      ...                ...   ...    ...     ...  
8437       e                N

In [63]:
len(all_df_merged["food_commodity_item"].unique())

25

In [64]:
all_df_merged["food_commodity_item"].unique()

array(['olive oil', 'palm oil', 'date', 'orange', 'apple', 'apricot',
       'avocado', 'banana', 'fig', 'grape', 'pear', 'pineapple',
       'watermelon', 'olive', 'currant', 'barley', 'millet', 'oat', 'rye',
       'sorghum', 'wheat', 'sunflower seed', 'spinach', 'artichoke',
       'wine'], dtype=object)

13 elements in common between the 3 datasets T-T
26 in commons when removing s.

In [71]:
def df_without_common(df ,column_name):
    return df[~df[column_name].isin(list(all_df_merged["food_commodity_item"]))]

def export_to_txt_the_column(df ,column_name :str,export_name :str):
    df = df_without_common(df ,column_name).drop_duplicates(subset = column_name)
    df[column_name].to_csv(f'data/{export_name}.txt', sep=" ", index=False)


export_to_txt_the_column(df_prod ,"Item","df_prod")
#export_to_txt_the_column(df_co2 ,"food_commodity_item","df_co2")
#export_to_txt_the_column(df_water ,"food_commodity_item","df_water")

In [66]:
export_to_txt_the_column(df_prod ,"Item","df_prod")

In [70]:
df_to_export_co2 =df_without_common(df_co2 ,"food_commodity_item")
df_to_export_co2 = df_to_export_co2.sort_values(by= "food_commodity_item",axis=0, ascending=True)

In [ ]:
df_without_common(df_co2 ,"food_commodity_item").to_excel("df_co2.xlsx")
df_without_common(df_water ,"food_commodity_item").to_excel("df_water.xlsx")

PermissionError: [Errno 13] Permission denied: 'df_co2.xlsx'

In [ ]:
common = pd.read_csv("food-commodity-mapping-csv(1).txt")

In [ ]:
common = common[~common["prod.txt"].str.contains("-")]

In [ ]:
common.to_excel("data/commmon.xlsx")

In [ ]:
from_co2_to_common = dict(zip(common['co2_item.txt'], common['Shortest Name']))
df_co2["food_commodity_item"] = df_co2["food_commodity_item"].replace(from_co2_to_common)
df_co2

Unnamed: 0    food_commodity_group                food_commodity_item  \
0             0  AGRICULTURAL PROCESSED                        BEER IN CAN   
1             1  AGRICULTURAL PROCESSED                      BEER IN GLASS   
2             2  AGRICULTURAL PROCESSED                   BEER MODULAR CAN   
3             3  AGRICULTURAL PROCESSED  CHOCOLATE OR CREAM FILLED COOKIES   
4             4  AGRICULTURAL PROCESSED                     SIMPLE COOKIES   
..          ...                     ...                                ...   
319         319                 FISHING                            OCTOPUS   
320         320                 FISHING                              SQUID   
321         321                 FISHING                     PRAWNS/SHRIMPS   
322         322                 FISHING                        OCTOPUS (F)   
323         323                 FISHING                 PRAWNS/SHRIMPS (F)   

     CF_food_item uncertainty          Suggested CF value  \
0        0.693000           L                     OK item   
1        0.960500           L                     OK item   
2        0.530600           L                     OK item   
3        1.533000           L                     OK item   
4        1.390750           L                     OK item   
..            ...         ...                         ...   
319      5.800000           H  better typol. or subtypol.   
320      6.910000           H  better typol. or subtypol.   
321      7.040000           L                     OK item   
322      7.878611           H             better typology   
323     15.065000           L                     OK item   

    food_commodity_typology  CF_food_typology  \
0                      BEER              0.69   
1                      BEER              0.69   
2                      BEER              0.69   
3                  BISCUITS              1.47   
4                  BISCUITS              1.47   
..                      ...               ...   
319               SHELLFISH              5.67   
320               SHELLFISH              5.67   
321               SHELLFISH              5.67   
322        SHELLFISH FROZEN             11.47   
323        SHELLFISH FROZEN             11.47   

               Food_commodity_sub-TYPOLOGY  CF_food_sub_typology  
0                                      NaN                   NaN  
1                                      NaN                   NaN  
2                                      NaN                   NaN  
3                                      NaN                   NaN  
4                                      NaN                   NaN  
..                                     ...                   ...  
319                   CEPHALOPOD (SUB-TYP)                  5.80  
320                   CEPHALOPOD (SUB-TYP)                  5.80  
321  CRUSTACEAN (SUB-TYP) (Prawns/Shrimps)                  7.04  
322                                    NaN                   NaN  
323                                    NaN                   NaN  

[324 rows x 10 columns]

In [ ]:
from_prod_to_common = dict(zip(common['prod.txt'], common['Shortest Name']))
df_prod["food_commodity_item"] = df_prod["food_commodity_item"].replace(from_co2_to_common)

KeyError: 'food_commodity_item'

In [ ]:
#Question quand dans un des df c'est 2 sous catégorie ou prend la moyenne puis on développe après?